### PR TITLE
added trending topics api

### PR DESCRIPTION
**Pull request by denizetkar**
**I added the controller, service, repository etc. for returning trending topics as a list. Although I could not find any error in my code, intellij gave errors when I tried to run the twitter project:**

2018-04-07 19:03:10.928 ERROR 9704 --- [cat-startStop-1] org.apache.catalina.core.ContainerBase   : A child container failed during start

java.util.concurrent.ExecutionException: org.apache.catalina.LifecycleException: Failed to start component [StandardEngine[Tomcat].StandardHost[localhost].TomcatEmbeddedContext[]]
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122) [na:na]
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191) [na:na]
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:941) ~[tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:872) [tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150) [tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1421) [tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1411) [tomcat-embed-core-8.5.29.jar:8.5.29]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) [na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1135) [na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) [na:na]
	at java.base/java.lang.Thread.run(Thread.java:844) [na:na]
Caused by: org.apache.catalina.LifecycleException: Failed to start component [StandardEngine[Tomcat].StandardHost[localhost].TomcatEmbeddedContext[]]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:167) [tomcat-embed-core-8.5.29.jar:8.5.29]
	... 6 common frames omitted
Caused by: org.apache.catalina.LifecycleException: Failed to start component [Pipeline[StandardEngine[Tomcat].StandardHost[localhost].TomcatEmbeddedContext[]]]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:167) [tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5125) ~[tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150) [tomcat-embed-core-8.5.29.jar:8.5.29]
	... 6 common frames omitted
Caused by: org.apache.catalina.LifecycleException: Failed to start component [org.apache.catalina.authenticator.NonLoginAuthenticator[]]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:167) [tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.core.StandardPipeline.startInternal(StandardPipeline.java:182) ~[tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150) [tomcat-embed-core-8.5.29.jar:8.5.29]
	... 8 common frames omitted
Caused by: java.lang.NoSuchMethodError: javax.servlet.ServletContext.getVirtualServerName()Ljava/lang/String;
	at org.apache.catalina.authenticator.AuthenticatorBase.startInternal(AuthenticatorBase.java:1186) ~[tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150) [tomcat-embed-core-8.5.29.jar:8.5.29]
	... 10 common frames omitted

2018-04-07 19:03:10.929 ERROR 9704 --- [           main] org.apache.catalina.core.ContainerBase   : A child container failed during start

java.util.concurrent.ExecutionException: org.apache.catalina.LifecycleException: Failed to start component [StandardEngine[Tomcat].StandardHost[localhost]]
	at java.base/java.util.concurrent.FutureTask.report(FutureTask.java:122) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.get(FutureTask.java:191) ~[na:na]
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:941) ~[tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:262) [tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150) [tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.core.StandardService.startInternal(StandardService.java:422) [tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150) [tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:793) [tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150) [tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.startup.Tomcat.start(Tomcat.java:367) [tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.springframework.boot.web.embedded.tomcat.TomcatWebServer.initialize(TomcatWebServer.java:107) [spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at org.springframework.boot.web.embedded.tomcat.TomcatWebServer.<init>(TomcatWebServer.java:86) [spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at org.springframework.boot.web.embedded.tomcat.TomcatServletWebServerFactory.getTomcatWebServer(TomcatServletWebServerFactory.java:409) [spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at org.springframework.boot.web.embedded.tomcat.TomcatServletWebServerFactory.getWebServer(TomcatServletWebServerFactory.java:174) [spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.createWebServer(ServletWebServerApplicationContext.java:179) [spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.onRefresh(ServletWebServerApplicationContext.java:152) [spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:544) [spring-context-5.0.5.RELEASE.jar:5.0.5.RELEASE]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:140) [spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:759) [spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:395) [spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:327) [spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1255) [spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1243) [spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at com.cmpe352_group1.twitter_project.TwitterProjectApplication.main(TwitterProjectApplication.java:10) [classes/:na]
Caused by: org.apache.catalina.LifecycleException: Failed to start component [StandardEngine[Tomcat].StandardHost[localhost]]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:167) [tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1421) ~[tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1411) ~[tomcat-embed-core-8.5.29.jar:8.5.29]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1135) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:844) ~[na:na]
Caused by: org.apache.catalina.LifecycleException: A child container failed during start
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:949) ~[tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:872) ~[tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150) [tomcat-embed-core-8.5.29.jar:8.5.29]
	... 6 common frames omitted

2018-04-07 19:03:10.931  INFO 9704 --- [           main] o.apache.catalina.core.StandardService   : Stopping service [Tomcat]
2018-04-07 19:03:10.932  WARN 9704 --- [           main] ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.context.ApplicationContextException: Unable to start web server; nested exception is org.springframework.boot.web.server.WebServerException: Unable to start embedded Tomcat
2018-04-07 19:03:10.945  INFO 9704 --- [           main] ConditionEvaluationReportLoggingListener : 

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2018-04-07 19:03:10.951 ERROR 9704 --- [           main] o.s.boot.SpringApplication               : Application run failed

org.springframework.context.ApplicationContextException: Unable to start web server; nested exception is org.springframework.boot.web.server.WebServerException: Unable to start embedded Tomcat
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.onRefresh(ServletWebServerApplicationContext.java:155) ~[spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:544) ~[spring-context-5.0.5.RELEASE.jar:5.0.5.RELEASE]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:140) ~[spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:759) [spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:395) [spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:327) [spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1255) [spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1243) [spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at com.cmpe352_group1.twitter_project.TwitterProjectApplication.main(TwitterProjectApplication.java:10) [classes/:na]
Caused by: org.springframework.boot.web.server.WebServerException: Unable to start embedded Tomcat
	at org.springframework.boot.web.embedded.tomcat.TomcatWebServer.initialize(TomcatWebServer.java:126) ~[spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at org.springframework.boot.web.embedded.tomcat.TomcatWebServer.<init>(TomcatWebServer.java:86) ~[spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at org.springframework.boot.web.embedded.tomcat.TomcatServletWebServerFactory.getTomcatWebServer(TomcatServletWebServerFactory.java:409) ~[spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at org.springframework.boot.web.embedded.tomcat.TomcatServletWebServerFactory.getWebServer(TomcatServletWebServerFactory.java:174) ~[spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.createWebServer(ServletWebServerApplicationContext.java:179) ~[spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.onRefresh(ServletWebServerApplicationContext.java:152) ~[spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	... 8 common frames omitted
Caused by: org.apache.catalina.LifecycleException: Failed to start component [StandardServer[-1]]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:167) ~[tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.startup.Tomcat.start(Tomcat.java:367) ~[tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.springframework.boot.web.embedded.tomcat.TomcatWebServer.initialize(TomcatWebServer.java:107) ~[spring-boot-2.0.1.RELEASE.jar:2.0.1.RELEASE]
	... 13 common frames omitted
Caused by: org.apache.catalina.LifecycleException: Failed to start component [StandardService[Tomcat]]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:167) ~[tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:793) ~[tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150) ~[tomcat-embed-core-8.5.29.jar:8.5.29]
	... 15 common frames omitted
Caused by: org.apache.catalina.LifecycleException: Failed to start component [StandardEngine[Tomcat]]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:167) ~[tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.core.StandardService.startInternal(StandardService.java:422) ~[tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150) ~[tomcat-embed-core-8.5.29.jar:8.5.29]
	... 17 common frames omitted
Caused by: org.apache.catalina.LifecycleException: A child container failed during start
	at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:949) ~[tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:262) ~[tomcat-embed-core-8.5.29.jar:8.5.29]
	at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150) ~[tomcat-embed-core-8.5.29.jar:8.5.29]
	... 19 common frames omitted

Process finished with exit code 1

### CODE REVIEW

File: twitter project/src/main/java/com/cmpe352_group1/twitter_project/TwitterProjectApplication.java [hunk 1]
```diff
@@ -6,7 +6,7 @@
 @SpringBootApplication
 public class TwitterProjectApplication {
 
-	public static void main(String[] args) {
-		SpringApplication.run(TwitterProjectApplication.class, args);
-	}
```
Comment (akinilerle) [inline twitter project/src/main/java/com/cmpe352_group1/twitter_project/TwitterProjectApplication.java]: There are no changes here.

File: twitter project/src/main/java/com/cmpe352_group1/twitter_project/core/GlobalExceptionHandler.java [hunk 2]
```diff
@@ -1,26 +1,24 @@
...
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+
+    @ExceptionHandler
+    @ResponseBody
+    public ResponseEntity indibuExceptionHandler(ProjectException exception){
+        return new ResponseEntity<>(exception.getMessage(), exception.getHttpStatus());
+    }
+
```
Comment (akinilerle) [inline twitter project/src/main/java/com/cmpe352_group1/twitter_project/core/GlobalExceptionHandler.java]: There is no change here neither.

File: twitter project/src/main/java/com/cmpe352_group1/twitter_project/model/ProjectException.java [hunk 3]
```diff
@@ -1,25 +1,24 @@
...
+
+import org.springframework.http.HttpStatus;
+
+public class ProjectException extends RuntimeException {
+
+    HttpStatus httpStatus;
+
+    public ProjectException(String message, HttpStatus httpStatus) {
+        super(message);
+        this.httpStatus = httpStatus;
```
Comment (akinilerle) [inline twitter project/src/main/java/com/cmpe352_group1/twitter_project/model/ProjectException.java]: No change

File: twitter project/src/main/java/com/cmpe352_group1/twitter_project/data/TrendingTopicsEntity.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.Date;
+
+@Entity
+public class TrendingTopicsEntity {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Long id;
+    private String name;
+    private String query;
```
Comment (akinilerle) [inline twitter project/src/main/java/com/cmpe352_group1/twitter_project/data/TrendingTopicsEntity.java]: There is no need for the query field

File: twitter project/src/main/java/com/cmpe352_group1/twitter_project/model/TrendingTopic.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
+package com.cmpe352_group1.twitter_project.model;
+
+public class TrendingTopic {
```
Comment (akinilerle) [inline twitter project/src/main/java/com/cmpe352_group1/twitter_project/model/TrendingTopic.java]: We do not need a second TT class

File: twitter project/src/main/java/com/cmpe352_group1/twitter_project/service/ProjectServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+@Service
+public class ProjectServiceImpl implements ProjectService {
+
+    @Autowired
+    TrendingTopicsRepository trendingTopicsRepository;
+
+    @Override
+    public List<TrendingTopic> getTT(Date date) {
+        // return trendingTopicsRepository.findByDate(date).stream().map(TrendingTopicsEntity::getTrendingTopic).collect(Collectors.toList());
```
Comment (akinilerle) [inline twitter project/src/main/java/com/cmpe352_group1/twitter_project/service/ProjectServiceImpl.java]: Please delete unnecessary comments

File: twitter project/src/main/java/com/cmpe352_group1/twitter_project/service/ProjectServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+        return new ArrayList<TrendingTopic>(
+                Arrays.asList(
+                        new TrendingTopic("#ChainedToTheRhythm", "%23ChainedToTheRhythm"),
+                        new TrendingTopic("#ChainedToTheRhythm2", "%23ChainedToTheRhythm2")
```
Comment (akinilerle) [inline twitter project/src/main/java/com/cmpe352_group1/twitter_project/service/ProjectServiceImpl.java]: This function should call the repository. Please push it after it is finished.

File: twitter project/src/test/java/com/cmpe352_group1/twitter_project/TwitterProjectApplicationTests.java [hunk 8]
```diff
@@ -9,8 +9,8 @@
 @SpringBootTest
 public class TwitterProjectApplicationTests {
 
-	@Test
-	public void contextLoads() {
-	}
+    @Test
```
Comment (akinilerle) [inline twitter project/src/test/java/com/cmpe352_group1/twitter_project/TwitterProjectApplicationTests.java]: No change

File: twitter project/.gitignore [hunk 9]
```diff
@@ -0,0 +1,25 @@
+/target/
+!.mvn/wrapper/maven-wrapper.jar
+
+### STS ###
+.apt_generated
```
Comment (akinilerle) [inline twitter project/.gitignore]: This project should be under webserver directory.

**Comment by akinilerle** [comment]
Also, bugs should be fixed before merging

**Comment by denizetkar** [comment]
With the commit 29567c2e65b7ab2203e62df5fb000916922562b4 , I fixed the above problem. However this time there seems to be something missing about mongodb that I have not been able to figure out yet. Despite receiving the following exception, application seems to respond to the http requests unlike before:

com.mongodb.MongoSocketOpenException: Exception opening socket
	at com.mongodb.connection.SocketStream.open(SocketStream.java:62) ~[mongodb-driver-core-3.6.3.jar:na]
	at com.mongodb.connection.InternalStreamConnection.open(InternalStreamConnection.java:126) ~[mongodb-driver-core-3.6.3.jar:na]
	at com.mongodb.connection.DefaultServerMonitor$ServerMonitorRunnable.run(DefaultServerMonitor.java:114) ~[mongodb-driver-core-3.6.3.jar:na]
	at java.base/java.lang.Thread.run(Thread.java:844) [na:na]
Caused by: java.net.ConnectException: Connection refused: connect
	at java.base/java.net.DualStackPlainSocketImpl.waitForConnect(Native Method) ~[na:na]
	at java.base/java.net.DualStackPlainSocketImpl.socketConnect(DualStackPlainSocketImpl.java:85) ~[na:na]
	at java.base/java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:400) ~[na:na]
	at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:243) ~[na:na]
	at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:225) ~[na:na]
	at java.base/java.net.PlainSocketImpl.connect(PlainSocketImpl.java:148) ~[na:na]
	at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:402) ~[na:na]
	at java.base/java.net.Socket.connect(Socket.java:591) ~[na:na]
	at com.mongodb.connection.SocketStreamHelper.initialize(SocketStreamHelper.java:59) ~[mongodb-driver-core-3.6.3.jar:na]
	at com.mongodb.connection.SocketStream.open(SocketStream.java:57) ~[mongodb-driver-core-3.6.3.jar:na]
	... 3 common frames omitted

File: twitter project/webserver/pom.xml [hunk 10]
```diff
@@ -1,57 +1,60 @@
...
-		<plugins>
-			<plugin>
-				<groupId>org.springframework.boot</groupId>
-				<artifactId>spring-boot-maven-plugin</artifactId>
-			</plugin>
-		</plugins>
-	</build>
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
```
Comment (akinilerle) [inline twitter project/webserver/pom.xml]: Only the changes should be pushed

File: twitter project/webserver/src/main/java/com/cmpe352_group1/twitter_project/TwitterProjectApplication.java [hunk 11]
```diff
@@ -6,7 +6,7 @@
 @SpringBootApplication
 public class TwitterProjectApplication {
 
-	public static void main(String[] args) {
-		SpringApplication.run(TwitterProjectApplication.class, args);
-	}
+    public static void main(String[] args) {
+        SpringApplication.run(TwitterProjectApplication.class, args);
```
Comment (akinilerle) [inline twitter project/webserver/src/main/java/com/cmpe352_group1/twitter_project/TwitterProjectApplication.java]: There is no change here

File: twitter project/webserver/src/main/java/com/cmpe352_group1/twitter_project/core/GlobalExceptionHandler.java [hunk 12]
```diff
@@ -1,26 +1,24 @@
...
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+@ControllerAdvice
+public class GlobalExceptionHandler {
+
+    @ExceptionHandler
+    @ResponseBody
+    public ResponseEntity indibuExceptionHandler(ProjectException exception){
+        return new ResponseEntity<>(exception.getMessage(), exception.getHttpStatus());
```
Comment (akinilerle) [inline twitter project/webserver/src/main/java/com/cmpe352_group1/twitter_project/core/GlobalExceptionHandler.java]: There is no change

File: twitter project/webserver/src/main/java/com/cmpe352_group1/twitter_project/model/ProjectException.java [hunk 13]
```diff
@@ -1,25 +1,24 @@
...
+public class ProjectException extends RuntimeException {
+
+    HttpStatus httpStatus;
+
+    public ProjectException(String message, HttpStatus httpStatus) {
+        super(message);
+        this.httpStatus = httpStatus;
+    }
+
+    public ProjectException() {
```
Comment (akinilerle) [inline twitter project/webserver/src/main/java/com/cmpe352_group1/twitter_project/model/ProjectException.java]: No change

File: twitter project/webserver/src/test/java/com/cmpe352_group1/twitter_project/TwitterProjectApplicationTests.java [hunk 14]
```diff
@@ -9,8 +9,8 @@
 @SpringBootTest
 public class TwitterProjectApplicationTests {
 
-	@Test
-	public void contextLoads() {
-	}
+    @Test
+    public void contextLoads() {
```
Comment (akinilerle) [inline twitter project/webserver/src/test/java/com/cmpe352_group1/twitter_project/TwitterProjectApplicationTests.java]: There is no change here

**Pull request merged**